sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,574 @@
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
+        async () => {
+          return [confirmEnrolmentInstitutionCode];
+        };
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have invalid footer.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent.replace("3000001", "3000002");
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expectedResult.errors = [
+        `Error processing the file ${confirmEnrolmentResponseFile}. Error: The total count of detail records mentioned in the footer record does not match with the actual total details records count.`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should stop processing the ECE response file when one of the detail records have invalid data.", async () => {
+      // Arrange
+      const confirmEnrolmentInstitutionCode = "CONF";
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        confirmEnrolmentInstitutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+      // Mock the implementation to return the mocked institution code.
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
+        async () => {
+          return [confirmEnrolmentInstitutionCode];
+        };
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have invalid application number.
+      // Note: The disbursement id is already an invalid value in file
+      // due to the placeholder DISBNUMBER.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent.replace("APPLNUMBER", "          ");
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expectedResult.errors = [
+        "Invalid unique index number for the disbursement record, Invalid application number at line 1.",
```

Comment:
when we say the line, shouldn't the line number be the same as that of the original file?

Reply:
https://github.com/bcgov/SIMS/pull/1965#discussion_r1205926825